client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,287 @@
+        }
+        return teamTotals;
+      } else // Neither, ignore request
+        return null;
+    }
+  }
+
+  // Return a modified 'seat' array or object including the seat widgets (not just the seat ids) to actually be used.
+  getIncludedSeats() {
+    const seats = this.get('seats');
+    if(Array.isArray(seats) || seats === null) // Scoreboard just using seats
+       return [...widgetFilter(w => w.get('type') == 'seat' && (this.get('includeAllSeats') || w.get('player')) && (!seats || seats.includes(w.get('id'))))]
+    else if(seats !== null && typeof seats == 'object') { // Scoreboard using teams
+      const newSeats = {};
+      for (const team in seats) {
+        newSeats[team] = asArray(seats[team]);
+        newSeats[team] = [... widgetFilter(w => w.get('type') == 'seat' && (this.get('includeAllSeats') || w.get('player')) && newSeats[team].includes(w.get('id')))];
+      }
+      return newSeats;
+    } else // 'seats' property is not array or object, ignore it.
+      return null;
+  }
+
+  getTotal(array) {
+    let sum = array.reduce((partialSum, a) => partialSum + parseFloat(a), 0);
+    if(isNaN(sum))
+      sum = 0
+    return sum;
+  }
+
+  addRowToTable(parent, values, cellType = 'td') {
+    const tr = parent.insertRow();
+    const v = asArray(values);
+    tr.innerHTML = Array(values.length).fill(`<${cellType}></${cellType}>`).join('');
+    for (let i=0; i < values.length; i++)
+      $a(cellType, tr)[i].innerText = values[i];
+    return tr;
+  }
+
+  tableCreate(seats) {
+    /* This routine creates the HTML table for display in the scoreboard. It is
+     * complicated by the fact that the `seats` property can be either an array of
+     * seat IDs or an object whose keys are team names and each of whose values is an
+     * array of seat IDs.
+     * There are two major sections: the first computes the pScores array, which contains
+     * the array of scores, either for seats or for teams. The second section uses the
+     * pScores array to construct the HTML table.
+     * There are lots of other things going on, to get the totals line, round names, etc
+     * correct.
+     */
+    const showTotals = this.get('showTotals');
+    const scoreProperty = this.get('scoreProperty');
+    let sortField = this.get('sortField');
+    let usePlayerColors = this.get('usePlayerColors');
+    const colors = []; // Array of player colors
+
+    // Compute number of scoring rounds to show
+    let rounds = this.get('rounds'); // User-supplied round names
+    let includedSeats = seats; // Array of all relevant seats
+    if(typeof(seats)=='object') // Array of all seats for team use
+      includedSeats = Object.keys(seats).reduce((union,key) => union.concat(seats[key]), []);
+
+    let numRounds=0;
+    for (let i=0; i < includedSeats.length; i++) {
+      const score = includedSeats[i].get(scoreProperty);
+      if(Array.isArray(score) && score.length > numRounds)
+        numRounds = score.length;
+    }
```

Comment:
This could probably be written something like `let numRounds=Math.max(...includedSeats.map(s=>s.get(scoreProperty)))` but it's fine the way it is. My version is missing the array check which might not be needed or it makes it as verbose as your version.